[admin]/admin2/server/admin_server.lua

```diff
@@ -15,6 +15,66 @@ aInteriors = {}
 aStats = {}
 aReports = {}
 aWeathers = {}
+aWhowas = {}
+
+function saveToDatabaseWhowas(nick, serial, ip, timestamp)
+    if not nick or not serial or not ip then 
+        return 
+    end
+    if not timestamp then 
+        timestamp = getRealTime().timestamp 
+    end
+    db.exec("DELETE FROM whowas WHERE name=? AND serial=?;", nick, serial)
+    db.exec("INSERT INTO whowas(name,serial,ip,time) VALUES(?,?,?,?);", nick, serial, ip, timestamp)
+end
+
+function clearRowsWhowas()
+    for k,data in ipairs(aWhowas) do
+        saveToDatabaseWhowas(data[1], data[2], data[3], data[4])
+    end
+    aWhowas = {}
+end
+
+function insertIntoWhowas(nick, serial, ip, timestamp)
+    if not nick or not serial or not ip then 
+        return 
+    end
+    if not timestamp then 
+        timestamp = getRealTime().timestamp 
+    end
+    table.insert(aWhowas, {nick, serial, ip, timestamp})
+end
+
+function persistPlayerWhowasInfo(player) -- Ensures that the player's information is up to date, it also ensures the integrity of the database
```

Comment:
```suggestion
-- Ensures that the player's information is up to date, it also ensures the integrity of the database
function persistPlayerWhowasInfo(player)
```

Comments should be above the function